src/processor/tree/tree_wrapper.rs

```diff
@@ -79,35 +96,68 @@ impl TreeWrapper {
             hex::encode(root_hash)
         );
 
-        assert_eq!(root_hash.as_bytes(), block.new_state_root);
-        tracing::debug!("Successfully processed block {}", block.l2_block_number);
+        let root_hash_bytes = root_hash.as_bytes();
+        if root_hash_bytes == block.new_state_root {
+            tracing::debug!("Successfully processed block {}", block.l2_block_number);
+
+            Ok(root_hash)
+        } else {
+            tracing::error!(
+                "Root hash mismatch!\nLocal: {:?}\nPublished: {:?}",
+                root_hash_bytes,
+                block.new_state_root
+            );
```

Comment:
Another thing, how about encoding the root hashes as hex instead? It's how they are represented up on L1 so I think it makes more sense.
```suggestion
        if root_hash_bytes.as_bytes() == block.new_state_root {
            tracing::debug!("Successfully processed block {}", block.l2_block_number);

            Ok(root_hash)
        } else {
            tracing::error!(
                "Root hash mismatch!\nLocal: {:?}\nPublished: {:?}",
                hex::encode(root_hash),
                # Not sure if this one is stored in the same endianness.
                hex::encode(block.new_state_root)
            );
```

Reply:
yes, I remember being irritated by the different formats when debugging...